### ISSUE TITLE
Attention Mask

### ISSUE THREAD

**Issue by Janspiry** [raywzy/ICT#8]
Hi, thanks for your nice works.
There are some details that bothered me.  I would appreciate it if you could give me some advice.
1. BERT choice
    I noticed `--BERT` option was be used in all Transformer training and inference processes.  **Which situation we do not need to select this option**?
2. attention mask
    As the paper described, transformer model capture the unmasked information to predict the probability distribution of missing regions. ![image](https://user-images.githubusercontent.com/32922494/131639675-e781afe5-ef66-43d9-b35e-5e8e9ad86f72.png). But in the code `CausalSelfAttention` , I found model will capture information at all position, and attention filling mask does not be used except input occlusion. **How can it guarantee to just pay attention on unmasked information**?
3. auto-regressive 
  As far as I understand, the model generate all masked pixels by a end to end mode rather than auto-regressive mode. During the inference, the model generate one pixel each iteration to improve sampling quality. If it works like I said, **how do we guarantee the pixel quality** of the first masked position in each iteration during the inference process?

Thanks again.

**Comment by Janspiry** [comment]
Thanks for your reply. I completely understood 1 and 2. 
About question 3, I want to confirm the question with you. I  want to know about why we can select the pixel in **first masked position** each sampling like code in `sample_mask function` Line98, uitl.py , since the model generate all masked pixels a time which have no difference in mask region in training.  Is it more reasonable to select the position who have the highest/Top-K probability in the rest of mask region each sampling.

**Comment by raywzy** [comment]
> Thanks for your reply. I completely understood 1 and 2.
> About question 3, I want to confirm the question with you. I want to know about why we can select the pixel in **first masked position** each sampling like code in `sample_mask function` Line98, uitl.py , since the model generate all masked pixels a time which have no difference in mask region in training. Is it more reasonable to select the position who have the highest/Top-K probability in the rest of mask region each sampling.

Absolutely. You could try the mentioned sampling strategy to see if the performance will become better. Currently I just sample the token in the sequential order.

**Comment by Janspiry** [comment]
I see, thanks!
